ui-tests/src/test/java/io/syndesis/qe/pages/integrations/editor/add/steps/DataMapper.java

```diff
@@ -174,14 +174,50 @@ public void selectMapping(String mappingName, SelenideElement containerElement)
         List<String> path = Arrays.asList(mappingName.trim().split("\\."));
         final String parent = path.get(0);
 
-        path.forEach(s -> {
-            SelenideElement detailElement = containerElement.find(By.id(s)).shouldBe(visible);
+        boolean previousHadNullId = false;
+
+        for( int i = 0; i < path.size(); i++ ) {
+            String s = path.get(i);
+            SelenideElement detailElement = containerElement.find(By.id(s));
+
+            if (s == null || s.equals("")) {
```

Comment:
Could the `s` actually ever be null? The `String.split` javadoc does not say anything about that, only mentions empty strings.

Also, we currently `find(By.id(s))`, which would immediately throw `IllegalArgumentException` if `s` was null.